src/java/picard/sam/DownsampleSam.java

```diff
@@ -39,13 +48,21 @@
     @Option(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "The output, downsampled, SAM or BAM file to write.")
     public File OUTPUT;
 
+    @Option(shortName="S", doc="The downsampling strategy to use. See usage for discussion.")
+    public Strategy STRATEGY = Strategy.ConstantMemory;
+
     @Option(shortName = "R", doc = "Random seed to use if reproducibilty is desired.  " +
             "Setting to null will cause multiple invocations to produce different results.")
-    public Long RANDOM_SEED = 1L;
+    public Integer RANDOM_SEED = 1;
 
     @Option(shortName = "P", doc = "The probability of keeping any individual read, between 0 and 1.")
     public double PROBABILITY = 1;
```

Comment:
Yeah. I added a log.warn() when running with P=1 to keep @nh13 happy.  It'll run just fine, but the user will get yelled at.